emissions_factors/scripts/utils.py

```diff
@@ -0,0 +1,64 @@
+import csv
+from pathlib import Path
+import uuid
+
+
+def uuid_generate_v4():
+    """generate a version 4 UUID"""
+    return str(uuid.uuid4())
```

Comment:
Can you implement a seeded version of this (seed the random generator once when the script starts running and then use it to generate the UUID)?

See here: https://stackoverflow.com/a/50743795
You could use either [`faker`](https://github.com/joke2k/faker) or `hashlib` to achieve this as mentioned there. Hashing is a great way to turn string IDs that are consistent into UUIDv4's. Just note that UUIDv4 has [6 bits that are meant to be hardcoded](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29) and not randomly selected, so using faker might be better as that library handles it for you.